src/sentry/profiles/consumers/process/factory.py

```diff
@@ -18,32 +18,39 @@ def process_message(message: Message[KafkaPayload]) -> None:
     sampled = is_sampled(message.payload.headers)
 
     if sampled or options.get("profiling.profile_metrics.unsampled_profiles.enabled"):
-        b64encoded = b64encode(message.payload.value).decode("utf-8")
-        process_profile_task.delay(payload=b64encoded, sampled=sampled, compressed_profile=False)
+        b64encoded_uncompressed = b64encode(message.payload.value).decode("utf-8")
 
-        if random.random() < options.get("taskworker.try_compress.profile_metrics"):
+        if random.random() < options.get("taskworker.try_compress.profile_metrics.rollout"):
```

Comment:
No need to change this, but in the future you can use `sentry.options.rollout.in_random_rollout()` for these kinds of checks.